### ISSUE TITLE
[$1000] [HOLD for payment 2023-04-03] No profile pic is displayed when send or received money notification is received

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15423]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app in android and login with user A, minimize the app
2. Open the app in any other device and login with user B
3. Open user A report
4. Click on plus and send / request money (any amount or any currency is okay)
5. Observe the notification on android app
6. From device which is logged in with user B, send simple message to user A and observe the notification in android app

## Expected Result:
Profile pic should be displayed besides name in send / request money notification like it does for normal message notification

## Actual Result:
Profile pic area in notification is with expensify logo for send / request money notification
Same issue happens with sending a message that contains the mail id or website URL too.
## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native

**Version Number:** 1.2.76-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/221016734-6700986d-b895-443d-a26c-1b407a577e8d.mp4

<img src="https://user-images.githubusercontent.com/43996225/221016813-a280a025-95bf-44b2-befa-7a8e4a7fb33f.jpg" width="40%" height="40%">

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1677141627200119

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b8db7c6642eee0f3</li>
        <li>Upwork Job ID: 1630230021117202432</li>
        <li>Last Price Increase: 2023-03-27</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @puneetlath (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by puneetlath** [comment]
I was able to confirm the issue. But will likely need to be internal given it deals with notifications. 

<img width="359" alt="image" src="https://user-images.githubusercontent.com/1371996/221607800-adc158c4-ec6f-405c-8c60-cefb056893cf.png">

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01b8db7c6642eee0f3

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor Plus for review of internal employee PR - @rushatgabhane (`Internal`)

**Comment by MelvinBot** [comment]
@puneetlath, @rushatgabhane Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by puneetlath** [comment]
Still looking into this.

**Comment by hayata-suenaga** [comment]
@puneetlath I'd like to take this issue to be familiar with notification system. Do you mind if I assign myself?

**Comment by puneetlath** [comment]
Absolutely, it's all yours!

**Comment by hayata-suenaga** [comment]
Avatar picture is not being displayed when the message is long. Because the message for receive/send money is long, the avatar is being hidden. The problem seems to stemming from [`CustomNotificationProvider::applyMessageStyle`](https://github.com/Expensify/App/blob/080e34bd2a3ead99c8600dc7058598c05742d109/android/app/src/main/java/com/expensify/chat/customairshipextender/CustomNotificationProvider.java#L159). I'll investigate more.

<img width="432" alt="Screenshot 2023-03-10 at 11 27 38 AM" src="https://user-images.githubusercontent.com/98560306/224408665-f5990d7d-f070-4ead-84ec-45f17b740569.png">

**Comment by hayata-suenaga** [comment]
After reading through the code, I couldn't find any reason why this is happening. I believe the issue reported is a behavior defined by Android Notification API. I summarize what I found so far:

- Notification for android is written in native code (Java). We're extending Airship's `ReactNotificationProvider` to connect custom native code with RN. (Airship is the service we use to send push notifications. We use their client library, too)
- We're setting the avatar [here](https://github.com/Expensify/App/blob/1a990daf3e3931a49e2dc4a5522e89d7fb258f3d/android/app/src/main/java/com/expensify/chat/customairshipextender/CustomNotificationProvider.java#L180).
- We're using a standard notification template [`NotificationCompat.Builder`](https://developer.android.com/develop/ui/views/notifications/build-notification#SimpleNotification) to create notifications.
- We're using [`NotificationCompat.MessagingStyle`](https://developer.android.com/reference/androidx/core/app/NotificationCompat.MessagingStyle) to display a sequence of messages (there might be only a single message).
- Because we're just passing data to Android API and relying on it to display notification info, the issue might actually be a behavior defined by the Android API.
- Another reason I believe this is Android defined behavior is that the issue is observed with long text messages, too. (The OP also mentions this happens with messages with attachments). There is no difference in data received for short and long messages.

**Comment by puneetlath** [comment]
Wow, nice sleuthing! My recommendation would be that we close this out in that case. Perhaps we record this somewhere (in SO?) so we don't forget why this is.

**Comment by hayata-suenaga** [comment]
@Julesssss Thank you for checking this 🙇🏻‍♂️ I double checked and the `type` in the notification payload is `reportComment` for both simple text messages and money requests.

Part of payload for simple text message.
```
"title": "hayata+test1@expensifail.com",
  "type": "reportComment",
  "reportID": 8892140336359017,
  "reportAction": {
    "person": [
      {
        "type": "TEXT",
        "style": "strong",
        "text": "hayata+test1@expensifail.com"
      }
    ],
    "actorEmail": "hayata+test1@expensifail.com",
    "actorAccountID": 24,
    "message": [
      {
        "type": "COMMENT",
        "html": "hello",
        "text": "hello",
        "isEdited": false,
        "reactions": []
      }
    ],
```

Part of payload for money quest (from `pushNotificationEventCallback`).
```
"title": "hayata+test1@expensifail.com",
  "type": "reportComment",
  "reportID": 8892140336359017,
  "shouldScrollToLastUnread": true,
  "roomName": null,
  "onyxData": [
    {
      "onyxMethod": "merge",
      "key": "report_8892140336359017",
      "value": {
        "reportID": "8892140336359017",
        "maxSequenceNumber": 29,
        "lastVisibleActionCreated": "2023-03-13 17:32:17.215",
        "lastReadTime": "2023-03-13 17:07:52.820",
        "lastMessageText": "Requested $100.00 from Hayata",
        "lastActorEmail": "hayata+test1@expensifail.com"
      }
    },
```

do you think your fix for the separate issue will also fix this one?

**Comment by Julesssss** [comment]
Ah, it must be something else then. But regardless, I'm sure. My screenshot above shows the app working as expected. I've completely re-written how notifications are rendered on Android and we are no longer rendering IOU notifications in a separate notification.

**Comment by Julesssss** [comment]
Oops, I'll keep you assigned for the triage credit 👍

**Comment by MelvinBot** [comment]
@Julesssss @hayata-suenaga this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a>. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by Julesssss** [comment]
I have a[ PR in review](https://github.com/Expensify/App/pull/15857) that fixes this and may other notification issues.

**Comment by MelvinBot** [comment]
@Julesssss, @hayata-suenaga Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by Julesssss** [comment]
Should be resolved now. I just verified this issue on staging and [the profile image is shown](https://github.com/Expensify/App/pull/15857#issuecomment-1483263213).

**Comment by dhanashree-sawant** [comment]
Hi @Julesssss, I have gone through the messages above and I understand you were working on it already but by any means, if this issue triggered you to work on profile pic part then is it eligible for reporting bonus? If not then that's fine. Good to hear that it is resolved.

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@Julesssss / @hayata-suenaga] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@Julesssss / @hayata-suenaga] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@Julesssss / @hayata-suenaga] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [] Determine if we should create a regression test for this bug.
- [ ] [@Julesssss / @hayata-suenaga] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by dhanashree-sawant** [comment]
Hi @Julesssss , melvin bot has updated issue name to hold for payment, can you confirm once whether it is eligible for reporting bonus?

**Comment by Julesssss** [comment]
I checked off the boxes, this was not a regression... Working on the test case now. 

Hey @dhanashree-sawant, yep this is a valid report. Currently our automation makes it easy to miss these reporter bonuses when it's worked on internally, but hopefully we can improve this and stop missing them soon 😬

**Comment by MelvinBot** [comment]
Triggered auto assignment to @puneetlath (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @Santhosh-Sellavel (`External`)

**Comment by MelvinBot** [comment]
Current assignees @Julesssss and @hayata-suenaga are eligible for the External assigner, not assigning anyone new.

**Comment by Julesssss** [comment]
Hey @puneetlath, I resolved this myself but could you please pay out the reporting bonus to @dhanashree-sawant? Also, while this wasn't a regression, I'd like to propose these new tests: 

_________

**1) Verify that messages longer than 115 chars are not being truncated**
- Android/iOS only
- From a separate client, send this message to the user:
    - this is a very very very very very very very very very very very very very very very very very very very very very very long message that wasn't previously possible! 😮😮😮😮😮😮😮😮
- The full message should be displayed

**2) Verify that messages and 1:1 IOU requests are displayed correctl**y
- From a separate client, send any message to the user
- From a separate client, send an IOU request to the Android user for any amount
- Both messages should be displayed correctly (ignoring the currency symbol issue)
- Both messages should be part of the same thread/notification (this only applies to newer Android versions)

**3) Verify that bill split messages are displayed correctly**
- From a separate client, create a new group that includes the user
- Send any message to the group
- It should be received by the android user
- Create a bill split between the group
- It should be received by the android user and formatted correctly 
- These will not be threaded, as they are from two separate reports

**Comment by puneetlath** [comment]
@dhanashree-sawant can you please apply here: https://www.upwork.com/jobs/~01b8db7c6642eee0f3

**Comment by dhanashree-sawant** [comment]
@puneetlath thanks, I have applied to the job
